sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py

```diff
@@ -0,0 +1,300 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from datetime import datetime
+from typing import List, Optional
+import uuid
+from azure.core.tracing.decorator import distributed_trace
+from azure.communication.rooms._models import (
+    CommunicationRoom,
+    RoomParticipant,
+    ParticipantsCollection
+)
+from azure.communication.rooms._shared.models import CommunicationIdentifier
+from ._generated._client import AzureCommunicationRoomsService
+from ._generated.models import (
+    CreateRoomRequest,
+    UpdateRoomRequest,
+    RemoveParticipantsRequest,
+    AddParticipantsRequest,
+    UpdateParticipantsRequest,
+    RoomJoinPolicy
+)
+
+from ._shared.utils import parse_connection_str, get_authentication_policy
+from ._version import SDK_MONIKER
+from ._api_versions import DEFAULT_VERSION
+
+class RoomsClient(object): # pylint: disable=client-accepts-api-version-keyword
+    """A client to interact with the AzureCommunicationService Rooms gateway.
+
+    This client provides operations to manage rooms.
+
+    :param str endpoint:
+        The endpoint url for Azure Communication Service resource.
+    :param str credential:
+        The access key we use to authenticate against the service.
+    """
+    def __init__(
+            self, endpoint: str,
+            credential: str,
```

Comment:
The `credential` parameter should never be a string.
If token auth is supported, it should be `TokenCredential`, or if _only_ key auth is supported, it should be `AzureKeyCredential`